src/tickit_devices/eiger/eiger_adapters.py

```diff
@@ -24,6 +24,78 @@ def command_404(key: str) -> str:
 
 LOGGER = logging.getLogger("EigerAdapter")
 
+_changed_parameters = {  # if not given, changed parameter list for key is [key]
+    "auto_summation": ["auto_summation", "frame_count_time"],
+    "count_time": [
+        "bit_depth_image",
+        "bit_depth_readout",
+        "count_time",
+        "countrate_correction_count_cutoff",
+        "frame_count_time",
+        "frame_time",
+    ],
+    "flatfield": ["flatfield", "threshold/1/flatfield"],
+    "frame_time": [
+        "bit_depth_image",
+        "bit_depth_readout",
+        "count_time",
+        "countrate_correction_count_cutoff",
+        "frame_count_time",
+        "frame_time",
+    ],
+    "incident_energy": [
+        "element",
+        "flatfield",
+        "incident_energy",
+        "photon_energy",
+        "threshold/1/energy",
+        "threshold/1/flatfield",
+        "threshold/2/energy",
+        "threshold/2/flatfield",
+        "threshold_energy",
+        "wavelength",
+    ],
+    "photon_energy": [
+        "element",
+        "flatfield",
+        "incident_energy",
+        "photon_energy",
+        "threshold/1/energy",
+        "threshold/1/flatfield",
+        "threshold/2/energy",
+        "threshold/2/flatfield",
+        "threshold_energy",
+        "wavelength",
+    ],
+    "pixel_mask": ["pixel_mask", "threshold/1/pixel_mask"],
+    "roi_mode": ["count_time", "frame_time", "roi_mode"],
+    "threshold/1/energy": [
+        "flatfield",
+        "threshold/1/energy",
+        "threshold/1/flatfield",
+        "threshold/2/flatfield",
+        "threshold_energy",
+    ],
+    "threshold/1/flatfield": ["flatfield", "threshold/1/flatfield"],
+    "threshold/1/mode": ["threshold/1/mode", "threshold/difference/mode"],
+    "threshold/1/pixel_mask": ["pixel_mask", "threshold/1/pixel_mask"],
+    "threshold/2/energy": [
+        "flatfield",
+        "threshold/1/flatfield",
+        "threshold/2/energy",
+        "threshold/2/flatfield",
+    ],
+    "threshold/2/mode": ["threshold/2/mode", "threshold/difference/mode"],
+    "threshold_energy": [
+        "flatfield",
+        "threshold/1/energy",
+        "threshold/1/flatfield",
+        "threshold/2/flatfield",
+        "threshold_energy",
+    ],
+    "threshold/difference/mode": ["difference_mode"],  # replicating API inconsistency
+}
+
```

Comment:
The code below would be neater if there was a function (somewhere in `tickit_devices.eiger.eiger` perhaps) that takes the put param and returns either a set from this map, or the parameter itself. Then just test that these two methods call serialize with the return of that function, and test the two cases of that simple function to get the coverage up.

Reply:
Okay, have moved to a function in eiger.eiger and added a test, checking that the put_config/put_threshold_config calls return the right encoded list for a few parameters

Reply:
Looks really neat now!